source/_localCaptioner/modelDownloader.py

```diff
@@ -0,0 +1,395 @@
+# -*- coding: UTF-8 -*-
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2025 NV Access Limited, tianze
+# This file may be used under the terms of the GNU General Public License, version 2 or later, as modified by the NVDA license.
+# For full terms and any additional permissions, see the NVDA license file: https://github.com/nvaccess/nvda/blob/master/copying.txt
+"""
+Multi‑threaded model downloader
+==============================
+
+Download ONNX / tokenizer assets from *Hugging Face* (or any HTTP host)
+with progress callbacks.
+
+Run this file directly for a CLI demonstration.
+
+Example
+-------
+.. code:: bash
+
+	python download_models.py  # will download to ``models`` sibling dir
```

Comment:
I don't think this should be a script run from the command line, I think it needs to be called directly from NVDA via python

Reply:
if you want to be able to test things more easily, you can call functions  from the NVDA python console